lib/widgets/profile.dart

```diff
@@ -0,0 +1,272 @@
+          child: Text(text,
+            style: TextStyle(color: const HSLColor.fromAHSL(1, 200, 1, 0.4).toColor()))));
+    } else {
+      return Text(text);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final userFields = sortAndAnnotateUserFields(user.profileData!, store.customProfileFields);
+    if (userFields.isNotEmpty) {
+      return DefaultTextStyle.merge(
+        style: const TextStyle(fontSize: 16, height: 1),
+        textAlign: TextAlign.left,
+        child: Column(
+          children: [
+            const SizedBox(height: 16),
+            Table(
+              border: TableBorder.all(style: BorderStyle.none),
+              columnWidths: const {
+                0: FixedColumnWidth(116),
+                1: FlexColumnWidth()},
+              defaultVerticalAlignment: TableCellVerticalAlignment.top,
+              children: userFields.map((e) => TableRow(
+                children: [
+                  Padding(
+                    padding: const EdgeInsets.fromLTRB(0, 4, 8, 4),
+                    child: Text(e.label, style: const TextStyle(fontWeight: FontWeight.bold, ))),
+                  renderProfileDataField(context, e, store),
+                ])).toList()),
+          ]));
+    } else {
+      return const SizedBox.shrink();
+    }
+  }
+}
+
+class ProfilePage extends StatelessWidget {
+  final int userId;
+
+  const ProfilePage({super.key, required this.userId});
+
+  static Route<void> buildRoute({required BuildContext context, required int userId}) {
+    return MaterialAccountWidgetRoute(context: context,
+      page: ProfilePage(userId: userId));
+  }
+
+  Widget buildErrorPage() {
+    return Scaffold(
+      appBar: AppBar(title: const Text("Error")),
+      body: const SingleChildScrollView(
+        child: Padding(
+          padding: EdgeInsets.all(16),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Icon(Icons.error),
+              Text("Could not show user profile."),
+            ]))));
+  }
+
+  void _createDmNavigation (context) {
+    final store = PerAccountStoreWidget.of(context);
+    final allRecipientIds = [store.account.userId, userId];
+    allRecipientIds.sort();
+    final narrow = DmNarrow(
+      selfUserId: store.account.userId,
+      allRecipientIds: allRecipientIds);
+    Navigator.push(context,
+      MessageListPage.buildRoute(context: context, narrow: narrow));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final user = store.users[userId];
+    if (user != null) {
+      final avatar = Avatar(userId: userId, size: 200, borderRadius: 200 / 8);
+      final Widget profileDataTable;
+      if (user.profileData != null) {
+        profileDataTable = ProfileDataTable(user:user);
+      } else {
+        profileDataTable = const SizedBox.shrink();
+      }
```

Comment:
I think this condition would be appropriate to push down inside `ProfileDataTable.build`, simplifying this code in the parent.